components/mediation/extensions/org.wso2.micro.integrator.transport.handlers/src/main/java/org/wso2/micro/integrator/transport/handlers/utils/SwaggerUtils.java

```diff
@@ -0,0 +1,302 @@
+                    pathParameter.required(true);
+                    operation.addParameter(pathParameter);
+                } else if (resourceParameterType
+                        .equals(AxisResourceParameter.ParameterType.QUERY_PARAMETER) && "GET".equals(method)) {
+                    //  Currently handling query parameter only for GET requests.
+                    QueryParameter queryParameter = new QueryParameter();
+                    queryParameter.setName(resourceParameter.getParameterName());
+                    queryParameter.setType(resourceParameter.getParameterDataType());
+                    queryParameter.required(true);
+                    operation.addParameter(queryParameter);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method will create the info section of the swagger document.
+     *
+     * @param dataServiceName name of the data-service.
+     * @param transports      enabled transports.
+     * @param serverConfig    Server config object.
+     * @param swaggerDoc      Swagger document object.
+     * @throws AxisFault Exception occured while getting the host address from transports.
+     */
+    private static void addSwaggerInfoSection(String dataServiceName, List<String> transports,
+                                              MIServerConfig serverConfig, Swagger swaggerDoc) throws AxisFault {
+
+        swaggerDoc.basePath("/" + SwaggerProcessorConstants.SERVICES_PREFIX + "/" + dataServiceName);
+
+        if (transports.contains("https")) {
+            swaggerDoc.addScheme(Scheme.HTTPS);
+            swaggerDoc.addScheme(Scheme.HTTP);
+            swaggerDoc.setHost(serverConfig.getHost("https"));
+        } else {
+            swaggerDoc.addScheme(Scheme.HTTP);
+            swaggerDoc.setHost(serverConfig.getHost("http"));
+        }
+
+        Info info = new Info();
+        info.title(dataServiceName);
+        info.setVersion("1.0");
+        info.description("API Definition of dataservice : " + dataServiceName);
+        swaggerDoc.setInfo(info);
+
+        swaggerDoc.addConsumes("application/json");
+        swaggerDoc.addConsumes("application/xml");
+
+        swaggerDoc.addProduces("application/json");
+        swaggerDoc.addProduces("application/xml");
+    }
+
+    /**
+     * Util method to fetch a swagger resource from the registry.
+     *
+     * @param resourcePath registry path to the resource.
+     * @return Swagger definition as string or null if error occurred.
+     */
+    public static String fetchSwaggerFromRegistry(String resourcePath) {
+        String defString = null;
+        SynapseConfiguration synapseConfig =
+                SynapseConfigUtils.getSynapseConfiguration(Constants.SUPER_TENANT_DOMAIN_NAME);
+        Registry registry = synapseConfig.getRegistry();
+        OMNode regContent = registry.lookup(resourcePath);
+
+        if (regContent instanceof OMText) {
+            defString = ((OMText) regContent).getText();
+        }
+
```

Comment:
What if regContent is not an instance of OMText? We will silently return null from here. Shouldn't we thrown an error instead?

Reply:
In the swagger for API feature by default, a swagger is created and added to the registry. So it's not an error according to implementations. However, for DSS I added a new custom exception.